### PR TITLE
OCPNODE-1632: Implement ImagePolicy and ClusterImagePolicy

### PR DESCRIPTION
**Pull request by QiWang19**
**- What I did**
Follow the design from the [enhancement](https://github.com/openshift/enhancements/blob/master/enhancements/api-review/add-ClusterImagePolicy-and-ImagePolicy-for-Signature-Verification.md). Add the implementation to ContainerruntimConfig controller for `policy.json` configuration. 
**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Comment by QiWang19** [comment]
ready for review.

File: pkg/controller/container-runtime-config/helpers.go [hunk 1]
```diff
@@ -753,3 +783,214 @@ func convertICSPToIDMS(icsp *apioperatorsv1alpha1.ImageContentSourcePolicy) *api
...
+		data, err := json.Marshal(policyObj)
+		if err != nil {
+			return nil, fmt.Errorf("error marshalling policy json for namespaced policies: %w", err)
+		}
+		namespaceJSONs[namespace] = data
+	}
+	return namespaceJSONs, nil
+}
+
+func imagePolicyConfigFileList(namespaceJSONs map[string][]byte, sigstoreRegistriesConfigYaml []byte) []generatedConfigFile {
```
Comment (mtrmac) [inline pkg/controller/container-runtime-config/helpers.go:980]: CRI-O has to this point no ability to correctly implement per-namespace policies (that has stalled in https://github.com/cri-o/cri-o/pull/7046 , and my further work on that is tracked in https://issues.redhat.com/browse/RUN-1980 , so isn’t this rather premature?

Of course the code proposed in this PR could be reviewed concurrently, but I don’t think it makes sense to merge it.

Comment (saschagrunert) [reply pkg/controller/container-runtime-config/helpers.go:980]: @mtrmac thank you for the clarification! Do you have plans to continue the CRI-O work (RUN-1980) in the near future?

Comment (mtrmac) [reply pkg/controller/container-runtime-config/helpers.go:980]: I’m definitely interested seeing this through to completion, after all the effort, but I don’t think I can spare the time in the next ~month at least.

Comment (mtrmac) [reply pkg/controller/container-runtime-config/helpers.go:980]: Note https://issues.redhat.com/browse/OTA-1170 ; AFAICS that would benefit from MCO implementing `ClusterImagePolicy` .

Comment (saschagrunert) [reply pkg/controller/container-runtime-config/helpers.go:980]: I agree that sounds like a plan. If we can deliver the namespaced policies in CRI-O v1.30 (late April) then we can pull those into the MCO implementation afterwards.

Focusing on the cluster policy sounds totally reasonable for now. :+1:

Comment (QiWang19) [reply pkg/controller/container-runtime-config/helpers.go:980]: @mtrmac @saschagrunert PR https://github.com/openshift/machine-config-operator/pull/4160 implementing ClusterImagePolicy is ready for review.

**Pull request closed without merging**